support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepExecution.java

```diff
@@ -54,6 +58,7 @@ public boolean start() throws Exception {
         actions.add(new CauseAction(new Cause.UpstreamCause(invokingRun)));
         List<ParameterValue> parameters = step.getParameters();
         if (parameters != null) {
+            completeDefaultParameters(parameters, (Job) project);
```

Comment:
:bug: return a new `List<ParameterValue>` rather than modifying the state of the `Step`.